### DISCUSSION TITLE
Convert metamask public key to format required by next.id proof service

### DISCUSSION THREAD

**Discussion by javaspeak** [DimensionDev/Maskbook#11090] [Q&A]
I have been trying to convert metamask's public key to a format required by next.id proof service and am failing.

My code looks like this:

```
      if (publicKeyBase64) {
        const publicKeyRaw = Buffer.from(publicKeyBase64, 'base64');
        const publicKeyHex = publicKeyRaw.toString('hex');

        const proofPayloadResponse: ProofPayloadResponse =
          await getProofPayloadResponse(twitterHandle, publicKeyHex);

        return proofPayloadResponse;
      }
```
     
The public key came from:

```
     const publicKeyBase64 = await window.ethereum.request({
        method: 'eth_getEncryptionPublicKey',
        params: [accounts],
     });
```

When I send the payload I get response:

  `  {"message":"public key not recognized"}`

Can anyone point me in the right direction in mask's code base so I can see how you did it.  Or paste some code here.

Have been grappling with next.id lack of typescript examples using metamask, so I thought maybe you mask guys have sorted this public key conversion problem already.

Thank you for your time and help.

**Comment by Jack-Works** [comment]
I have no idea, but if you're dealing with EC key, you may be interested in https://www.npmjs.com/package/@masknet/base, which exports ECKeyIdentifier (packages/base/src/Identifier/ec-key.ts) and has a `.publicKeyAsHex` property. That property may help. Also cc @guanbinrui

**Reply by javaspeak** [reply]
Thank you for your answer.  Will try it out.
